src/__experimental__/components/date/date-picker.component.js

```diff
@@ -110,10 +110,16 @@ function getDisabledDays(minDate, maxDate) {
   }
 
   if (minDate) {
+    if (!DateHelper.isValidDate(minDate)) {
+      return null;
+    }
     days.push({ before: DateHelper.stringToDate(minDate) });
   }
 
   if (maxDate) {
+    if (!DateHelper.isValidDate(maxDate)) {
+      return null;
+    }
```

Comment:
if we have a valid `maxDate` and invalid `minDate` then the valid date will not be pushed. Is that the expected behaviour?

Reply:
it could be simplified to:
```
const hasValidMinDate = minDate && DateHelper.isValidDate(minDate);
const hasValidMaxDate = maxDate&& DateHelper.isValidDate(maxDate);

if (hasValidMinDate) {
  days.push({ before: DateHelper.stringToDate(minDate) });
}
if (hasValidMaxDate ) {
  days.push({ after: DateHelper.stringToDate(maxDate) });
}
```

Reply:
Yes you are correct 🤦‍♂ , I'm going to remove this as it doesn't actually fix the issue and raise it as a separate ticket